### PR TITLE
Update translation documentation for 1.7.8

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Updated translation documentation including new features from 1.7.8

### CODE REVIEW

File: src/content/1.7/modules/creation/module-translation/new-system.md [hunk 1]
```diff
@@ -214,25 +230,15 @@ For instance, translating the string "Welcome to this page!" can be done like th
...
-```twig
-{{ 'Hello %username%!'|trans({'%username%': 'John'}, 'Modules.Mymodule.Admin') }}
+```smarty
+{l s='Hello %username%!' sprintf=['%username%' => 'John'] d='Modules.Mymodule.Somefile'} }}
 ```
 
 ## Translating your module
 
-To be translatable through the new translation interface, modules must opt-in. This can be done by declaring this function on your module's main class:
+Modules must opt-in to become translatable using the new Back Office translation interface. This can be done by declaring the following function on your module's main class:
```
Comment (sowbiba) [inline src/content/1.7/modules/creation/module-translation/new-system.md:241]: We dont look at this opt-in when requesting the modules in the "Modify translations" form

Comment (eternoendless) [reply src/content/1.7/modules/creation/module-translation/new-system.md:241]: This toggles access to the new translation interface. If the module hasn't opted in, you get the legacy translation interface ([see TranslationRouteFinder](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/Adapter/Translations/TranslationRouteFinder.php#L130))

Comment (sowbiba) [reply src/content/1.7/modules/creation/module-translation/new-system.md:241]: Ok. It's done on the routing level. Nothing to do when feeding the translator ?

**Review by sowbiba — APPROVED** [review]

File: src/content/1.7/development/internationalization/translation/native-module-translation.md [hunk 2]
```diff
@@ -1,157 +1,19 @@
...
-the "Translations in PrestaShop 1.5" chapter, variables are to be marked
-using `sprintf()` markers, such as `%s` or `%1$s`.
-
-Making "Hello %s!" translatable words in easy: we just need to use this
-code:
-
-```
-{l s='Hello %s!' sprintf=$my_module_name d='Modules.MyModule'}
-```
+In PrestaShop lingo, "native modules" refers to all the modules that are bundled and distributed with the Core.
```
Comment (Progi1984) [inline src/content/1.7/development/internationalization/translation/native-module-translation.md:15]: `lingo` ?

Comment (sowbiba) [reply src/content/1.7/development/internationalization/translation/native-module-translation.md:15]: Yeah I searched it too. That means language or something like that

Comment (sowbiba) [reply src/content/1.7/development/internationalization/translation/native-module-translation.md:15]: https://dictionary.cambridge.org/fr/dictionnaire/anglais/lingo

File: src/content/1.7/development/internationalization/translation/using-the-translator.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+### Outside controllers
+
+If you are outside a controller, and after careful consideration you think you absolutely need some stuff translated, then you can add it as a dependency of your class:
+
+```php
+<?php
+// SomeService.php
+
+namespace PrestaShop\PrestaShop\Core\Foo\Bar;
```
Comment (Progi1984) [inline src/content/1.7/development/internationalization/translation/using-the-translator.md:53]:
```suggestion
// SomeService.php

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Foo\Bar;
```

File: src/content/1.7/development/internationalization/translation/using-the-translator.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+class SomeService
+{
+    private $translator;
```
Comment (Progi1984) [inline src/content/1.7/development/internationalization/translation/using-the-translator.md:59]:
```suggestion
    /**
     * @var TranslatorInterface
     */
    private $translator;
```

File: src/content/1.7/development/internationalization/translation/using-the-translator.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    ```
+
+## Twig templates
+
+In `.twig` files, you can use the `trans` filter from Twig:
+
+```twig
+<div>{{ 'Sort by'|trans({}, 'Admin.Actions') }}</div>
+```
+
```
Comment (Progi1984) [inline src/content/1.7/development/internationalization/translation/using-the-translator.md:116]: An example with replacements ?

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: src/content/1.7/development/internationalization/translation/using-the-translator.md (hunk 3 above)
Comment (eternoendless) [reply src/content/1.7/development/internationalization/translation/using-the-translator.md:53]: Do you think this is relevant for this example?

File: src/content/1.7/development/internationalization/translation/using-the-translator.md (hunk 4 above)
Comment (eternoendless) [reply src/content/1.7/development/internationalization/translation/using-the-translator.md:59]: Not relevant for the example IMO

File: src/content/1.7/development/internationalization/translation/using-the-translator.md (hunk 5 above)
Comment (eternoendless) [reply src/content/1.7/development/internationalization/translation/using-the-translator.md:116]: Okay

File: src/content/1.7/development/internationalization/translation/using-the-translator.md (hunk 3 above)
Comment (Progi1984) [reply src/content/1.7/development/internationalization/translation/using-the-translator.md:53]: As users/readers copy/paste code, always put good practices in sample code.

**Comment by Progi1984** [comment]
@eternoendless I think your rebase has added out of scope commits in this PR. Isn't it ?

**Comment by eternoendless** [comment]
> @eternoendless I think your rebase has added out of scope commits in this PR. Isn't it ?

True, that's weird

File: src/content/1.7/development/internationalization/translation/using-the-translator.md (hunk 3 above)
Comment (eternoendless) [reply src/content/1.7/development/internationalization/translation/using-the-translator.md:53]: It makes the example longer to read and understand, it drives the focus away from the subject.

Comment (Progi1984) [reply src/content/1.7/development/internationalization/translation/using-the-translator.md:53]: As you want.

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @eternoendless

**Pull request merged**